user-data-store-server/src/main/java/com/wultra/security/userdatastore/controller/PhotoController.java

```diff
@@ -0,0 +1,107 @@
+import com.wultra.security.userdatastore.service.PhotoService;
+import io.getlime.core.rest.model.base.request.ObjectRequest;
+import io.getlime.core.rest.model.base.response.ObjectResponse;
+import io.getlime.core.rest.model.base.response.Response;
+import io.swagger.v3.oas.annotations.Operation;
+import jakarta.validation.constraints.NotBlank;
+import jakarta.validation.constraints.Size;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.Optional;
+
+/**
+ * REST controller providing API for CRUD for photos.
+ *
+ * @author Roman Strobl, roman.strobl@wultra.com
+ */
+@RestController
+@Validated
+@Slf4j
+class PhotoController {
+
+    private final PhotoService photoService;
+
+    @Autowired
+    PhotoController(PhotoService photoService) {
+        this.photoService = photoService;
+    }
+
+    /**
+     * Return photos for the given user and document.
+     *
+     * @param userId user identifier
+     * @param documentId document identifier
+     * @return photos
+     */
+    @Operation(
+            summary = "Return photos",
+            description = "Return photos for the given user and document."
+    )
+    @GetMapping("/photos")
+    public ObjectResponse<PhotoResponse> fetchPhotos(@NotBlank @Size(max = 255) @RequestParam String userId, @NotBlank @Size(max = 255) @RequestParam String documentId) {
+        logger.info("Fetching photos for document ID: {}", documentId);
+        final PhotoResponse photos = photoService.fetchPhotos(userId, Optional.ofNullable(documentId));
+        return new ObjectResponse<>(photos);
+    }
+
+    /**
+     * Create a photo for the given user and document.
+     *
+     * @param request Photo create request
+     * @return photo create response
+     */
+    @Operation(
+            summary = "Create a photo",
+            description = "Create a photo for the given user and document."
+    )
+    @PostMapping("/admin/photos")
+    public ObjectResponse<PhotoCreateResponse> createPhoto(@RequestBody final ObjectRequest<PhotoCreateRequest> request) {
+        logger.info("Creating photo for user ID: {}", request.getRequestObject().userId());
+        final PhotoCreateResponse response = photoService.createPhoto(request.getRequestObject());
+        return new ObjectResponse<>(response);
+    }
+
+    /**
+     * Delete photos for the given user and document.
+     *
+     * @param userId user identifier
+     * @param documentId document identifier
+     * @return response
+     */
+    @Operation(
+            summary = "Delete photos",
+            description = "Delete photos for the given user and document."
+    )
+    @DeleteMapping("/admin/photos")
+    public Response deletePhotos(@NotBlank @Size(max = 255) @RequestParam String userId, @NotBlank @Size(max = 255) @RequestParam(required = false) String documentId) {
```

Comment:
Not blank constraint considers null as invalid value. Apparently there isn't predefined annotation to validate "null or not blank".

Reply:
I will add the `@NotNull` annotations, I missed that.

Reply:
No, I mean `documentId` argument is not required and at the same time has not blank constraint. I think the NotBlank annotation should be removed.

Reply:
Ah, in this case yes.

Reply:
Fixed now.